### PR TITLE
fix(REST): Save otherLicenseIds while patching Release - 1735

### PR DESCRIPTION
**Pull request by akapti**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.

1. Fix the `otherLicenseIds` not being saved while patching release via REST API issue.
2. Updated the REST API documentation accordingly.

Issue: #1735 

### How To Test?
Update the Release via REST API with following scenario:

1. Add valid `otherLicenseIds` (present in sw360 db) field in request body and update the release -> Update should be successful and `otherLicenseIds` should contain all the licenses added in update request.
2. Add empty `otherLicenseIds` field in request body and update the release -> Update should be successful and `otherLicenseIds` field should be empty.
3. Add invalid (not present in sw360 db) `otherLicenseIds` field in request body and update the release -> Update should fail with error: `License with ids [license-id1, license-id2] does not exist in SW360 database.`

### Checklist
Must:
- [x] All related issues are referenced in commit messages and in PR

Signed-off-by: akapti <abdul.kapti@siemens-healthineers.com>

### CODE REVIEW

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/JacksonCustomizations.java [hunk 1]
```diff
@@ -458,8 +458,8 @@ static abstract class ComponentMixin extends Component {
                 "setAdditionalData",
                 "mainLicenseIdsIterator",
                 "setBinaryDownloadurl",
-                "setBinaryDownloadurl",
```
Comment (heliocastro) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/JacksonCustomizations.java:461]: Is this line removed correctly ?

**Review by heliocastro — CHANGES_REQUESTED** [review]
Please improve the changelog with the number of the issue associated. In this case was isse 1735

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/JacksonCustomizations.java (hunk 1 above)
Comment (akapti) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/JacksonCustomizations.java:461]: There were two occurrences of `setBinaryDownloadurl`, so I removed one, and replace it with `subscribersIterator`.

**Review by heliocastro — CHANGES_REQUESTED** [review]
The git changelog still not mention the issue
Please add the proper issue in the git changelog, not only on Github entry

**Comment by afsahsyeda** [comment]
Hi @akapti ,
Testing was successful.

**Review by smrutis1 — APPROVED** [review]
Code looks good.

**Pull request merged**